embassy-embedded-hal/src/shared_bus/i2c.rs

```diff
@@ -27,6 +27,7 @@ use core::future::Future;
 
 use embassy::blocking_mutex::raw::RawMutex;
 use embassy::mutex::Mutex;
+#[cfg(feature = "nightly")]
```

Comment:
perhaps it's better to completely cfg out the entire async `shared_bus/i2c.rs` and `shared_bus/spi.rs` files? so that the types don't appear at all in public API, instead of being there but with no impls at all, which could be confusing.

Reply:
Yeah i was thinking the same but was not sure in what mod it would make sense to put the Errors... In the `shared_bus` root perhaps?

Reply:
perhaps have `shared_bus::asynch` and `shared_bus::blocking`, and the errors in `shared_bus`?